__tests__/createRetryFetch.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+const createRetryFetch = require('../src/createRetryFetch');
+const createTimeoutFetch = require('../src/createTimeoutFetch');
+const TimeoutError = require('../src/TimeoutError');
+
+function resolveInMs(ms) {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve('Fetch Resolved'), ms);
+  });
+}
+
+describe('createRetryFetch', () => {
+  it('should resolve if the response finishes before the default timeout', async () => {
+    const mockFetch = jest.fn(() => resolveInMs(100));
+    const anotherEnhancedFetch = createTimeoutFetch(500);
+    const enhancedFetch = createRetryFetch(anotherEnhancedFetch)(mockFetch);
+    const promise = enhancedFetch('/test');
+    await expect(promise).resolves.toBe('Fetch Resolved');
+  });
```

Comment:
adapted to remove the `enhancedFetch` argument:
```suggestion
  it('should resolve if the response finishes before the default timeout', async () => {
    const mockFetch = jest.fn(() => resolveInMs(100));
    const enhancedFetch = createRetryFetch()(mockFetch);
    await expect(enhancedFetch('/test')).resolves.toBe('Fetch Resolved');
  });
```
etc